tests/test_macros.py

```diff
@@ -0,0 +1,92 @@
+import dataclasses
+import os
+
+import dbt.tracking
+import pytest
+from _pytest.fixtures import SubRequest
+from dbt.adapters.factory import get_adapter, register_adapter, AdapterContainer
+from dbt.clients.jinja import MacroGenerator
+from dbt.config.runtime import RuntimeConfig
+from dbt.context import providers
+from dbt.contracts.connection import ConnectionState
+from dbt.contracts.graph.manifest import Manifest
+from dbt.adapters.spark.connections import (
+    SparkConnectionManager,
+    PyodbcConnectionWrapper,
+)
+from dbt.parser.manifest import ManifestLoader
+from dbt.tracking import User
+from pyspark.sql import SparkSession
+from sodaspark.scan import Connection
+
+
+dbt.tracking.active_user = User(os.getcwd())
+
+
+class _SparkConnectionManager(SparkConnectionManager):
```

Comment:
I have opened an [issue](https://github.com/dbt-labs/dbt-spark/issues/272) in `dbt-spark` to add a `pyspark` connection to the package. Then it is not needed to do this anymore.

Reply:
Really interesting, responded in the issue